source/fe/fe_poly_tensor.cc

```diff
@@ -358,24 +373,30 @@ FE_PolyTensor<PolynomialType, dim, spacedim>::fill_fe_values(
 
   // Create table with sign changes, due to the special structure of the RT
   // elements.
+
   // TODO: Preliminary hack to demonstrate the overall prinicple!
 
   // Compute eventual sign changes depending on the neighborhood
   // between two faces.
   std::fill(fe_data.sign_change.begin(), fe_data.sign_change.end(), 1.0);
 
-  if (mapping_type == mapping_raviart_thomas)
-    internal::FE_PolyTensor::get_face_sign_change_rt(cell,
-                                                     this->dofs_per_face,
-                                                     fe_data.sign_change);
-  else if (mapping_type == mapping_nedelec)
-    internal::FE_PolyTensor::get_face_sign_change_nedelec(cell,
-                                                          this->dofs_per_face,
-                                                          fe_data.sign_change);
+  internal::FE_PolyTensor::get_face_sign_change_rt(cell,
+                                                   *this,
+                                                   this->mapping_type,
+                                                   fe_data.sign_change);
+
+  internal::FE_PolyTensor::get_face_sign_change_nedelec(cell,
```

Comment:
This is clearer now. I wonder whether the two functions could be merged into one, though I'd leave this for a follow-up patch.

Reply:
I thought of that when I saw these side-by-side. If these were merged into one function, it would likely need to be a function that handles rt, nedelec, and is easy to modify in case future elements require something similar.

Reply:
Yes. Leave it as is for the moment. We can always come back to it.